app/database/queries/breeze_buddy/analytics.py

```
@@ -716,6 +738,75 @@ def get_analytics_lead_based_query(
+        picked AS (
+            SELECT request_id, MIN(attempt_no) AS pick_attempt
+            FROM attempts
+            WHERE outcome IS NOT NULL AND outcome NOT IN ('NO_ANSWER', 'BUSY')
+            GROUP BY request_id
+        )
+        SELECT
+            CASE WHEN pick_attempt >= 4 THEN '4+' ELSE pick_attempt::text END AS bucket,
+            COUNT(*) as count
+        FROM picked
+        GROUP BY 1;
+    """
```

Ruff (0.15.18)

[error] Possible SQL injection vector through string-based query construction

(S608)